### ISSUE TITLE
Bookmarks dissapears on swap documents

### ISSUE THREAD

**Issue by vsopko** [oleg-shilo/PyMap.VSIX#21]
https://github.com/oleg-shilo/PyMap.VSIX/assets/15255277/497367b5-65fb-4cee-9415-5077a2fefa50

**Comment by oleg-shilo** [comment]
Yeah, it's already fixed by 44dbd921416937db38719876927cb58eeff72f16

**Comment by oleg-shilo** [comment]
I do not want to publish another release on the marketplace and it does not support pre-releases so I only published it on GitHub so you can update your environment.

https://github.com/oleg-shilo/PyMap.VSIX/releases/tag/v2.2.2.1-pre

Will make it public when is used for a while and some feedback is collected.

**Comment by vsopko** [comment]
Reinstall to 2.2.2.1-pre. The issue is still exists.

**Comment by oleg-shilo** [comment]
Hm... I do not see it...
Can you share the file that triggers it. 
And maybe upload a new video. That one does not play...

**Comment by vsopko** [comment]
https://github.com/oleg-shilo/PyMap.VSIX/assets/15255277/3c92f4a5-f993-4f30-8902-82c5a2a8b366

**Comment by oleg-shilo** [comment]
Yeah, I found it. The problem was caused by a mistake in the bookmark validation mechanism overlooking the **members** bookmarks (your test case) while rook-level bookmarks were handled correctly (my test).
Thank you for the video, without it, it wouldn't be easy to pinpoint.

The defect is corrected now.

**Comment by oleg-shilo** [comment]
The fix is available here: 
https://github.com/oleg-shilo/PyMap.VSIX/releases/tag/v2.2.2.2-pre

**Comment by vsopko** [comment]
Reinstalled, now a new error appears: sometimes scroll bar margins are not visible when re-opening studio with documents already open (you need to close and reopen the document to display them). In addition, the scroll bar margins sometimes are not visible on new bookmarks, and sometimes are visible.

**Comment by oleg-shilo** [comment]
> now a new error appears:

No, it's not new. It was always there :)

This is something that I noticed too but I was not able to pinpoint the cause. 
It's not clear to me what triggers margin rendering. Thus when opening a fresh instance if VS with the CodeMap loaded correctly never shows the bookmarks on editor's vertical margin. Until the document is saved. This does trigger the rendering.

IDE eventing and rendering are beyond my expertise and currently, I have to fulfill some other project commitments so I am not going to look at it right now. Particularly because "margin support" was not in the intended scope of this effort. It was just that extra that is difficult to get right. But I may have a look at it if I get some time window for that.

I am more than happy if you do have time to look at this.

**Comment by vsopko** [comment]
> I am more than happy if you do have time to look at this.

Sorry, but I also don’t have the experience and time to fix this ( I will use it as is.
